src/node/state.ml

```diff
@@ -25,6 +25,10 @@ type t = {
   data_folder : string;
   pending_operations : timestamp Operation_map.t;
   block_pool : Block_pool.t;
+  (* TODO: we need a bound on the size of this and put
+     behind an abstract type. We should also change how
+     this works once we have an indexer. See https://github.com/marigold-dev/deku/issues/535 *)
+  applied_blocks : Block.t list;
```

Comment:
Naive question, why a list and not a Set?

Reply:
Semantically it's a more like set, but in this case switching to `Block.t Set.t` won't buy us any additional performance. The right thing to implement a data structure with an upper bound on size and that is indexed on the dimension we access it from (block level), like a `Block.t (Map.Make (Int64)).t`. But a better thing to do is to implement the indexer per the linked issue.

Reply:
>Semantically it's a more like set

Yes, that is exactly what I understood!

>in this case switching to Block.t Set.t won't buy us any additional performance

Is there any benchmark about `Set` vs `List` in OCaml?
What about using Array instead of List, I think I read something about Array is 7% more performant that List.

>But a better thing to do is to implement the indexer per the linked issue.

Definitively agreed.

Reply:
@gsebil08 Array will be more costly when we commit a block, a list makes sense regarding this.